### DISCUSSION TITLE
DOCKER_BUILDKIT=1 environment variable without reboot

### DISCUSSION THREAD

**Discussion by phinzphinz** [dokku/dokku#4690] [Q&A] [answered]
In the [documentation](https://dokku.com/docs~v0.24.10/deployment/builders/dockerfiles/) it says:
> If your Dockerfile is using Docker engine's buildkit (not to be confused with buildpacks), then the DOCKER_BUILDKIT=1 environment variable needs to be set. One way to do this is to edit /etc/environment on your dokku host and reboot your instance. 

What other ways exist to set the `DOCKER_BUILDKIT=1` environment variable for dokku without having to reboot my instance?

**Comment by phinzphinz** [comment, marked as answer]
I found a way: One can move the dokku executable `mv /usr/bin/dokku /usr/bin/dokkuraw` and write a bash script that sets the environment variable:
```
#!/bin/bash 
DOCKER_BUILDKIT=1 dokkuraw "$@"
```

Save this file as `/usr/bin/dokku` and then make it executable `chmod a+x /usr/bin/dokku`. Maybe this should be added to the documentation because for some people it is important to avoid a reboot.

**Reply by josegonzalez** [reply]
You can specify the variable via `dokku config:set --global DOCKER_BUILDKIT=1`, or any other method of setting global environment variables (`/etc/default` is another). No need to reboot anything.
